skypy/galaxy/tests/test_redshift.py

```diff
@@ -1,12 +1,82 @@
 import numpy as np
 
-from skypy.galaxy.redshift import smail, herbel_redshift, herbel_pdf
-from unittest.mock import patch
-from scipy.stats import kstest, norm
-import scipy.integrate
+from scipy.stats import kstest
+
+
+def test_schechter_lf_redshift():
+
+    from skypy.galaxy.redshift import schechter_lf_redshift, redshifts_from_comoving_density
+    from astropy.cosmology import FlatLambdaCDM
+
+    # fix this cosmology
+    cosmo = FlatLambdaCDM(H0=70, Om0=0.3)
+
+    m_lim = 25.
+    z = np.asarray([1., 2.])
+    M_star = -20
+    phi_star = 1e-3
+    fsky = 1/41253
+
+    # redshifts for blue and red galaxies
+    z_blu = schechter_lf_redshift(z, M_star, phi_star, -1.3, m_lim, fsky, cosmo, noise=False)
+    z_red = schechter_lf_redshift(z, M_star, phi_star, -0.5, m_lim, fsky, cosmo, noise=False)
+
+    # comoving densities computed manually
+    density = np.multiply(phi_star, [[0.68128448958519065, 0.0198597466483111],   # blue, z = 1, 2
+                                     [0.62047343466172848, 0.04958533460341455]]) # red, z = 1, 2
+
+    # turn into galaxies/surface area
+    density *= 4*np.pi*fsky*cosmo.differential_comoving_volume(z).to_value('Mpc3/sr')
+
+    # integrate total number
+    n_gal = np.trapz(density, z, axis=-1)
+
+    # make sure noise-free samples have right sizes
+    np.testing.assert_allclose([len(z_blu), len(z_red)], n_gal, atol=1.0)
+
+    # sample with noise this time
+    z_blu = schechter_lf_redshift(z, M_star, phi_star, -1.3, m_lim, fsky, cosmo, noise=True)
+    z_red = schechter_lf_redshift(z, M_star, phi_star, -0.5, m_lim, fsky, cosmo, noise=True)
+
+    # since only two points are given, check uniform distribution between z[0] and z[1]
+    D, p = kstest(z_blu, 'uniform', args=(z[0], z[1]-z[0]))
+    assert p > 0.01, 'D = {}, p = {}'.format(D, p)
+    D, p = kstest(z_red, 'uniform', args=(z[0], z[1]-z[0]))
+    assert p > 0.01, 'D = {}, p = {}'.format(D, p)
```

Comment:
Not sure if I like these tests. They only pass because our *implementation* gives a uniform approximation to the true pdf when `len(z)==2`. They don't test against any actual properties of the Schechter function itself e.g. alpha, M_star. And they don't test the implementation if len(z) > 2. But I'm struggling to come up with something better, any thoughts?

Reply:
They are failing now as well, so I am curious about any suggestions, too.